plugin_ide.core.tests/src-lang/melnorme/lang/ide/core/engine/StructureModelTest.java

```diff
@@ -60,8 +60,7 @@ public FixtureSourceModelManager() {
 			super(new DocumentReconcileManager(), new ProblemMarkerUpdater());
 		}
 		
-		public final SourceFileStructure DEFAULT_STRUCTURE = new SourceFileStructure(null, null, 
-			(Indexable<ParserError>) null);
+		public final SourceFileStructure DEFAULT_STRUCTURE = new SourceFileStructure(null, (Indexable<ParserError>) null);
```

Comment:
Yeah this is definitely wrong. You are adding a location field to every StructureElement in a tree, even though the location is the same for all in the tree. I've added: https://github.com/RustDT/RustDT/commit/fe73508dce4baba8844e7ac2762440c3fba18ffd